[qt5] Update to 5.15.2

Neumann-A opened the pull request:
TODO: ~~qt5-mqtt~~, ~~test qt5-webengine~~
~~(mqtt currently missing git tag)~~
closes #14675 

~~depends: #15049~~

~~Probably requires VM update for linux to be green again. (If somebody could figure out which tests are failing.)~~

~~Somebody needs to fix qt5-webengine again......the first error is easy the second one not so much~~

The error:
`ERROR: Feature 'xcb' was enabled, but the pre-condition 'features.thread && libs.xcb && tests.xcb_syslibs && features.xkbcommon-x11' failed.`
probably due to:
`test config.qtbase_gui.libraries.xcb_util FAILED`
~~so xcb_util needs to be installed within CI~~ (solved)

Neumann-A committed:
[qt5] Update to 5.15.2
(mqtt currently missing git tag)

Neumann-A committed:
deactivate update of ports

Neumann-A commented on ports/qt5-imageformats/CONTROL:
```diff
@@ -1,5 +1,5 @@
-Build-Depends:  qt5-base[core], tiff, libwebp, jasper
+Source: qt5-imageformats
+Version: 5.15.2
+Port-Version: 1
```
```suggestion
Port-Version: 0
```

Neumann-A committed:
update version

Neumann-A committed:
reduce port version

bl-ue commented on ports/qt5-base/cmake/qt_port_hashes.cmake:
```diff
@@ -1,55 +1,55 @@
+set(QT_HASH_qt5-doc                 ce2c003b37829da102f243ca271548cff0597b4b667109ca9533971a490b8b20eb3183af6e0b209ad58964f2be2e757f83933a3f8c484dd0814750e24d1a244e)
+set(QT_HASH_qt5-x11extras           beaefc865299f4e7c637baa96eb3a69adbe1a41fc7735b46cfec0768c9243391600e69078630fffb4aceba106b7512fd802e745abc38ddab5253233301c93ed9)
+set(QT_HASH_qt5-androidextras       cacd9252f313595d21eb6a94ffabbd5fff476af82aa619b4edfc77a9f405f52bd1b89da97c8f7dadf8c403035a243205a25a2f85250ebc0590bf68f914cdbf3a)
 #set(QT_HASH_qt5-canvas3d            0) deprectaed
```
```suggestion
#set(QT_HASH_qt5-canvas3d            0) deprecated

```

bl-ue commented on ports/qt5-base/cmake/qt_port_hashes_latest.cmake:
```diff
@@ -8,48 +8,48 @@ set(QT_PORT_LIST base 3d activeqt charts connectivity datavis3d declarative game
+set(QT_HASH_qt5-doc                 ce2c003b37829da102f243ca271548cff0597b4b667109ca9533971a490b8b20eb3183af6e0b209ad58964f2be2e757f83933a3f8c484dd0814750e24d1a244e)
+set(QT_HASH_qt5-x11extras           beaefc865299f4e7c637baa96eb3a69adbe1a41fc7735b46cfec0768c9243391600e69078630fffb4aceba106b7512fd802e745abc38ddab5253233301c93ed9)
+set(QT_HASH_qt5-androidextras       cacd9252f313595d21eb6a94ffabbd5fff476af82aa619b4edfc77a9f405f52bd1b89da97c8f7dadf8c403035a243205a25a2f85250ebc0590bf68f914cdbf3a)
 #set(QT_HASH_qt5-canvas3d            0) deprectaed
```
```suggestion
#set(QT_HASH_qt5-canvas3d            0) deprecated

```

Neumann-A committed:
fix typo

Neumann-A committed:
fix patch for linux missing pthread

Neumann-A committed:
fix qt5-tools

Neumann-A committed:
fix qt5-webengine

Neumann-A committed:
add xcb-util to linux provisioning script

Neumann-A committed:
fix version numbers

Neumann-A committed:
fix other typo

Neumann-A committed:
remove new hashes file

Neumann-A commented:
@JackBoosY @strega-nil could somebody investigate the problems `vcpkg_from_git` has with the official git of qt https://code.qt.io/cgit/qt/qtmqtt.git. qt5-mqtt is currently using a github mirror which hasn't yet updated to 5.15.2 so this PR will be blocked until then or the problem with `vcpkg_from_git` and the official git are resolved

JackBoosY commented:
@Neumann-A Seems 5.15.2 doesn't published on the github.
See https://github.com/qt/qtmqtt/tags

Neumann-A commented:
@JackBoosY:

> So as you said, we need to wait for the upstream publish that version.

we only need to wait if we use the github mirror. If we could use the official git at https://code.qt.io/cgit/qt/qtmqtt.git we wouldn't need to wait since it has the tag. The problem is that I wasn't able to get `vcpkg_from_git` to work with the official git repro. (Should probably be considered a vcpkg bug. I already ran into a few issues with `vcpkg_from_git` where it wasn't able to download from a git repro)

JackBoosY commented:
@Neumann-A  Maybe some parameters caused that. You can print the clone command and try it manually on you machine.

Neumann-A commented:
@JackBoosY. I already tried to make it work (e.g. by making --depth 1 and everything optional) but without success (I didn't try very long). `vcpkg_from_git` is doing a bit more than a simple `git clone`

Neumann-A commented:
@JackBoosY Also needs to have tag `depends_vm_update`

dweckmann commented on ports/qt5-base/cmake/qt_port_hashes_latest.cmake:
```diff
@@ -8,48 +8,48 @@ set(QT_PORT_LIST base 3d activeqt charts connectivity datavis3d declarative game
                  purchasing quickcontrols quickcontrols2 remoteobjects script scxml sensors serialport speech svg tools virtualkeyboard webchannel websockets
                  webview winextras xmlpatterns doc x11extras androidextras translations serialbus webengine webglplugin wayland)
 
-set(QT_HASH_qt5-base                40b687c046b25a6717834ffe2616ee4f373d75214ec1c7e3a26502cd0cde9f0a872eaee99f06c54c7a3625ae85df5cdd3a3b54a160e8e37c7cfeb5800d026fe4)
```
Suggested change :): 

4: set(QT_PATCH_VER 1) -> set(QT_PATCH_VER 2)

Neumann-A committed:
fix latest version info

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

# Conflicts:
#	ports/qt5-base/CONTROL

Neumann-A committed:
switch to vcpkg_from_git
adjust vcpkg_from_git to handle tags.
add OUT_REF for vcpkg_from_git to enable scriptable update of ports

Neumann-A committed:
regenerate docs

Neumann-A committed:
[vcpkg_from_git] new options TAG and OUT_REF
- TAG github tag to checkout
- OUT_REF github commit id related to tag or ref
(useable for automatic updates of ports if used with a version tag)

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	ports/qt5-base/CONTROL

Neumann-A committed:
Apply suggestions from code review

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

Neumann-A committed:
additional changes due to CR

Neumann-A committed:
regenerate docs

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into from_git_support_tags

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

Neumann-A committed:
Merge remote-tracking branch 'origin/from_git_support_tags' into update_qt_5_15_2

# Conflicts:
#	docs/maintainers/vcpkg_from_git.md
#	scripts/cmake/vcpkg_from_git.cmake

and update qt5-mqtt

Neumann-A commented:
@JackBoosY: remove depends-upstream changes. Changed it to depend on the other PR #15049

Neumann-A commented on ports/qt5-webengine/portfile.cmake:
```diff
@@ -37,7 +37,10 @@ vcpkg_add_to_path(PREPEND "${GPERF_DIR}")
 set(PATCHES common.pri.patch 
-            gl.patch)
+            gl.patch
+            build_1.patch
+            build_2.patch
+            build_3.patch)
```
Could be replaced with the upstream patches if somebody cares and give me a link to the correct patchset.

BillyONeal replied:
Speaking as the person who usually beats the "should be upstream thing" drum very hard given that future Qt will be qt6 and thus there are relatively little worries about these being "brittle" for the purposes of this port, I think these are OK as is.

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

vicroms commented:
/azp run

ras0219 commented on ports/qt5-location/CONTROL:
```diff
@@ -1,5 +1,5 @@
-Build-Depends: qt5-base[core], qt5-declarative, qt5-quickcontrols, qt5-quickcontrols2, qt5-serialport
+Source: qt5-location
+Version: 5.15.2
+Port-Version: 0
```
```suggestion
```

ras0219 commented on ports/qt5-location/rename_third_party_libs_conflicts.patch:
```diff
@@ -1,68 +0,0 @@
-diff --git a/src/3rdparty/clip2tri/clip2tri.pro b/src/3rdparty/clip2tri/clip2tri.pro
```
Just checking -- this was incorporated upstream?

Neumann-A replied:
yes. You already used the upstream patch here. 
alternativly you can check:
https://github.com/qt/qtlocation/blob/5.15.2/src/3rdparty/clip2tri/clip2tri.pro
https://github.com/qt/qtlocation/blob/5.15.2/src/3rdparty/clipper/clipper.pro
https://github.com/qt/qtlocation/blob/5.15.2/src/3rdparty/poly2tri/poly2tri.pro

Neumann-A committed:
Remove 'Port-Version:0' from qt5-location

Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

BillyONeal commented:
The mac build failed due to a transient infrastructure issue ( https://github.com/microsoft/vcpkg/pull/15584 ). `/azp run` after the other entries come back green should fix it.

BillyONeal commented:
/azp run

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

# Conflicts:
#	ports/qt5-base/CONTROL
#	ports/qt5/CONTROL

Neumann-A committed:
Merge branch 'update_qt_5_15_2' of https://github.com/Neumann-A/vcpkg into update_qt_5_15_2

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

# Conflicts:
#	ports/qt5-base/CONTROL

Neumann-A committed:
add all those versions

JackBoosY commented:
Can you please resolve the conflicts?

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_qt_5_15_2

# Conflicts:
#	port_versions/baseline.json
#	port_versions/q-/qt5.json
#	ports/qt5/CONTROL

Neumann-A committed:
fix versions

Neumann-A commented:
@JackBoosY You probably want to remove `depends:different-pr`

JackBoosY commented:
Let's merge this PR.

dan-shaw commented:
Thanks everyone!
